src/Jobs/Recruit.Vacancies.Jobs/Jobs/TransferVacanciesFromProviderJob.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Esfa.Recruit.Vacancies.Client.Application.Queues;
+using Esfa.Recruit.Vacancies.Client.Application.Queues.Messages;
+using Esfa.Recruit.Vacancies.Client.Domain.Repositories;
+
+namespace Esfa.Recruit.Vacancies.Jobs
+{
+    public class TransferVacanciesFromProviderJob
+    {
+        private readonly IVacancyQuery _vacanciesQuery;
+        private readonly IRecruitQueueService _queueService;
+
+        public TransferVacanciesFromProviderJob(IVacancyQuery vacanciesQuery, IRecruitQueueService queueService)
+        {
+            _vacanciesQuery = vacanciesQuery;
+            _queueService = queueService;
+        }
+
+        public async Task Run(long ukprn, long legalEntityId, Guid userRef, string userEmail, string userName)
```

Comment:
TransferReason should be passed in at this point and added to the message

Reply:
See above.

Reply:
👍